docs/guides/deployment/kubernetes.md

```diff
@@ -132,18 +134,83 @@ If you already have a registry, skip ahead to the next section.
+
+**Environment Variables**
+
+Add environment variables to set when starting a flow run.
+You can specify Python packages to install at runtime with `{"EXTRA_PIP_PACKAGES":"my_package"}`. For example `{"EXTRA_PIP_PACKAGES":"pandas==1.2.3"}` will install pandas version 1.2.3.
+Alternatively, you can specify package installation in a custom Dockerfile, which can allow you to take advantage of image caching.
+As we'll see below, Prefect can help us create a Dockerfile with our flow code and the packages specified in a `requirements.txt` file baked in.
+
+**Namespace**
+
+Set the Kubernetes namespace to create jobs within, such as `prefect`. By default, set to **default**.
+
+**Image**
+
+Specify the Docker container image for created jobs. If not set, the latest Prefect 2 image will be used.
+
+**Image Pull Policy**
+
+Select from the dropdown options to specify when to pull the image.
 
-Now let's switch over to Prefect Cloud, where we'll create a new work pool.
-Hit the plus button on the work pools page and choose Kubernetes from the list of options.
-On the next page, set Finished Job TTL to 60 so that completed flow runs are cleaned up, and set Pod Watch Timeout Seconds to 300, especially if you are using a __serverless__ type node pool since these tend to have longer startup times.
-You may also want to set a custom namespace, such as `prefect`.
-Generally you should leave the cluster config blank as the worker will already be provisioned with appropriate access and permissions.
-Finally, leave the image field blank as we'll override that in each deployment.
+**Finished Job TTL**
+
+You may want to set to 60 so that completed flow runs are cleaned up after a minute.
+
+**Pod Watch Timeout Seconds**
+
+Number of seconds for pod creation to complete before timing out.
+Consider setting to 300, especially if using a **serverless** type node pool, as these tend to have longer startup times.
+
+**Kubernetes Cluster Config**
+
+You can configure the Kubernetes cluster to use for job creation by specifying a `KubernetesClusterConfig` block.
+Generally you should leave the cluster config blank as the worker should be provisioned with appropriate access and permissions.
+
+!!! Note "Advanced Settings"
+  Want to modify the default base job template to add other fields or delete existing fields?
+
+  Select the **Advanced** tab and edit the JSON representation of the base job template.
+
+  For example, to set the namespace to `myorg` remove the namespace section under variables:
+
+    ```json
+      "namespace": {
+        "title": "Namespace",
+        "description": "The Kubernetes namespace to create jobs within.",
+        "default": "default",
+        "type": "string"
+      },
+    ```
+
+  and hardcode `"namespace": "myorg",` under `job_configuration->namespace` and `job_configuration->job_manifest->metadata->namespace`.
+
+  Running deployments with this work pool will now create flow runs in the `myorg` namespace.
```

Comment:
We should probably use a different example since namespace is already covered by the default template. CPU requests/limits would be a good common case.

I haven't tested, but something like this:

```suggestion
  For example, to set a CPU request, add the following section under variables:

    ```json
      "cpu_request": {
        "title": "CPU Request",
        "description": "The CPU allocation to request for this pod.",
        "default": "default",
        "type": "string"
      },
    ```

  Next add the following to the first `containers` item under `job_configuration`:

    ```json
            ...
            "containers": [
              {
                ...,
                "resources": {
                  "requests": {
                    "cpu": {{ cpu_request }}"
                  }
                }
              }
            ],
            ...
    ```

  Running deployments with this work pool will now request the specified CPU.
```